OpenRA.Mods.Common/Traits/Armament.cs

```diff
@@ -141,15 +147,20 @@ public Armament(Actor self, ArmamentInfo info)
 
 		public virtual WDist MaxRange()
 		{
-			return new WDist(Util.ApplyPercentageModifiers(Weapon.Range.Length, rangeModifiers));
+			return new WDist(Util.ApplyPercentageModifiers(Weapon.Range.Length, initialRangeModifiers));
```

Comment:
It wasn't cached for usage in CheckFire like you do now, but there is in Created self.TraitsImplementing<IRangeModifier>().**ToArray()**.Select(m => m.GetRangeModifier()) so they were cached for MaxRange(). What @RoosterDragon reports is that 'Select...' was evaluated for each MaxRange call what doesn't happen now. You could replace initialRangeModifiers with rangeModifiers.Select(...) to keep behaviour if needed.